File: Day 3/solve.py
def solvept1():
    total = 0
    for i in data:
        for j in i[:len(i)//2]:
            if j in i[len(i)//2:]:
                total += priority(j)
                break

    return total

def solvept2():
    total = 0
    i = 0
    while i <= len(data) - 3:
        first, second, third = data[i], data[i+1], data[i+2]
        for j in first:
            if j in second and j in third:
                total += priority(j)
                break
        i += 3

    return total

def priority(a):
    return ord(a) - 96 if ord(a) >= 97 else ord(a) - 38


if __name__ == "__main__":
    data = open("Day 3/day3input.txt", "r").readlines()
    print(solvept1(), solvept2())

File: Day 3/solve.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <algorithm>
using namespace std;


int priority(char a) {
    int c = a;
    // if (c >= 97) {
    //     return c - 96;
    // } else {
    //     return c - 38;
    // }

    return (c >= 97) ? c -96 : c - 38;
}

int solvept1(ifstream &a){
    string line;
    int total = 0;
    while (getline (a, line)) {
        string second_half = line.substr(line.length()/2, string::npos);
        for (int i = 0; i < line.length()/2; i++) {
            bool d = false;
            for (int j = 0; j < line.length()/2; j++) {
                if (line[i] == second_half[j]) {
                    total += priority(line[i]);
                    d = true;
                    break;
                }
            }
            if (d) break;
        }
    }
    cout << total << endl;
    return 0;
}


int main() {
    ifstream file("Day 3/day3input.txt");
    solvept1(file);
    file.close();
    return 0;
}